include/secp256k1.h

```diff
@@ -152,15 +152,37 @@ typedef int (*secp256k1_nonce_function)(
 # endif
 #endif
 #ifndef SECP256K1_API
-/* All cases not captured by the Windows-specific logic. */
-# if defined(__GNUC__) && (__GNUC__ >= 4) && defined(SECP256K1_BUILD)
-   /* Building libsecp256k1 using GCC or compatible. */
-#  define SECP256K1_API extern __attribute__ ((visibility ("default")))
-# else
-   /* Fall back to standard C's extern. */
-#  define SECP256K1_API extern
+# if defined(__GNUC__) && (__GNUC__ == 4) && defined(SECP256K1_BUILD)
+   /* Building libsecp256k1 using GCC or compatible on non-Windows. */
+#  if defined(SECP256K1_FORCE_HIDDEN_VISIBILITY)
+    /* Hidden visibility is requested explicitly.
+     *
+     * Forcing hidden visibility can be useful, e.g., when building a static
+     * library which is linked into a shared library, and the latter should not
+     * reexport the libsecp256k1 API. Since this will create an unusable shared
+     * library, SECP256K1_FORCE_HIDDEN_VISIBILITY should only be used when
+     * building a static library. (You want to ./configure with --disable-shared
+     * if using Autotools.)
+     *
+     * While visibility is a concept that applies only to shared libraries,
+     * setting visibility will still make a difference when building a static
+     * library: the visibility settings will be stored in the static library,
+     * solely for the potential case that the static library will be linked into
+     * a shared library. In that case, the stored visibility settings will
+     * resurface and be honored for the shared library. */
+#   define SECP256K1_API extern __attribute__ ((visibility("hidden")))
```

Comment:
Win32 needs this branch too, no? Otherwise it'll still require `-fvisibility=hidden` for static libs?

Reply:
Actually, maybe `#if defined(SECP256K1_FORCE_HIDDEN_VISIBILITY)` could just go at the top, and keep with the `#ifndef(SECP256K1_API)` pattern below it?

Reply:
No worries.